File: project_platypus/codegen/java/analysis.py
from dataclasses import dataclass
from enum import Enum, auto


class AnalysisPass(Enum):
    DEAD_CODE        = auto()
    DEOBFUSCATION    = auto()
    UNICODE_RECOVERY = auto()

@dataclass
class AnalysisConfig:
    enable_deobfuscation:    bool = True
    enable_unicode_recovery: bool = True
    enable_dead_code:        bool = True
    dead_code_algorithm:     str  = 'z'
    deobfuscation_level:     int  = 2 # 1 = safe, 2 = aggressive, 3 = speculative
    unicode_display:         str  = 'both'

File: experimental/rust/platypus-codegen/src/java/analysis.rs
/// Analysis configuration — translates codegen/java/analysis.py

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum AnalysisPass {
    DeadCode,
    Deobfuscation,
    UnicodeRecovery,
}

#[derive(Debug, Clone)]
pub struct AnalysisConfig {
    pub enable_deobfuscation:    bool,
    pub enable_unicode_recovery: bool,
    pub enable_dead_code:        bool,
    /// "z", "reachability", or "both"
    pub dead_code_algorithm:     String,
    /// 1 = safe, 2 = aggressive, 3 = speculative
    pub deobfuscation_level:     u8,
    /// "unicode", "escaped", or "both"
    pub unicode_display:         String,
}

impl Default for AnalysisConfig {
    fn default() -> Self {
        AnalysisConfig {
            enable_deobfuscation:    true,
            enable_unicode_recovery: true,
            enable_dead_code:        true,
            dead_code_algorithm:     "z".to_string(),
            deobfuscation_level:     2,
            unicode_display:         "both".to_string(),
        }
    }
}
